common/src/main/java/org/mvndaemon/mvnd/common/OsUtils.java

```diff
@@ -101,6 +102,29 @@ public static long findProcessRssInKb(long pid) {
         }
     }
 
+    public static String findJavaHomeFromPATH() {
+        final String basename = Os.current().isUnixLike() ? "java" : "java.exe";
+        final String PATH = System.getenv("PATH");
+        if (null == PATH) {
+            return null;
+        }
+        for (final String dirname : PATH.split(File.pathSeparator)) {
```

Comment:
This is not completely right.  If one of the path element contains an escaped reserved character such as `:`, this won't work.  The split should have to take that into account, maybe by using a regex that ensures that the `pathSeparator` is not preceded by an odd number of backslashes.

Reply:
In the latest commit, the env PATH is not parsed, and the `java` is auto searched by `ProcessBuilder`.